src/Aspire.Hosting/Dashboard/proto/resource_service.proto

```diff
@@ -155,6 +155,11 @@ message Resource {
 
     // The list of urls that this resource exposes
     repeated Url urls = 13;
+
+    // The style of the state. This is used to determine the state icon.
```

Comment:
Did you consider encoding a FluentUI icon name (e.g. `ErrorCircle`), style (e.g. `Filled`) and colour (e.g. `Blue`) here instead? That would give a lot more control over the UI to the server.

Reply:
I would prefer that actually. So just parse the enum?

Reply:
Looks like they're not enums: https://github.com/microsoft/fluentui-blazor/tree/main/src/Assets/FluentUI.Icons/Icons

This method might help, though it doesn't look cheap so results should be cached:

https://github.com/microsoft/fluentui-blazor/blob/main/src/Assets/FluentUI.Icons/Icons.cs#L20

You pass it an instance of [`IconInfo`](https://github.com/microsoft/fluentui-blazor/blob/dev/src/Core/Components/Icons/IconInfo.cs) which takes name, size and variant (filled/regular).

Reply:
I decided to stick with fixed names.